modules/fts/pages/fts-troubleshooting.adoc

```diff
@@ -0,0 +1,244 @@
++
+If the search clause is scoped to a field, FTS uses the same analyzer that the index mapping dictates for that field.
+
+* After analysis, what terms is the search looking for in the index?
++
+Once you know what analyzer is being used on your search query, use \http://analysis.blevesearch.com/ to see the exact terms that the analyzer outputs for your search.
+For example, if you’ve determined that the "en" analyzer is being used, a search for "marketing blunders" will result in term searches for "market" and "blunder."
+
+* Run a special query called a term search for one term at a time, exactly as the terms come from the analyzer in step 2.
+Be sure to return the document id match.
+Rerun for each term.
++
+----
+curl -XPOST -H "Content-Type: application/json" \
+        http://localhost:9200/api/index/perf_fts_index/query \
+        -d '{
+            "query": {
+                "conjuncts": [
+                    {
+                        "term": "market"
+                    },
+                    {
+                        "ids": ["Y"]
+                    }
+                ]
+            }
+        }'
+----
+
+* Did you get any results?
++
+If yes, the search query is working fine.
+If not, check the following:
+
+ ** The document (as analyzed) does not contain the terms in question.
+Review the contents of the document.
+ ** The terms in the document don’t analyze the way the user expected.
+Try changing the analyzer, possibly define a custom analyzer.
+ ** The index mapping may be incorrect and the expected analyzer isn’t actually being used.
+
+* Advanced: Digging deeper into the index
++
+Run command-line tools on the pindex that contains the document you expect to match.
+
+ $ cbft-bleve dump doc [path_to_pindex] [doc_id]
++
+This dumps everything the index knows about that document.
+
+== *Q: Why is this document in the search results, or why is this document ranked high in the search results?*
+
+Check the following:
+
+* Run the query with the option "explain" set to true.
+Understand which terms contributed to the document scoring the way it did.
+Higher numbers indicate that a term contributes more weight.
+* Experiment with boost to adjust this.
+For example, a query like "name:margaret^2 hobby:singing" would give twice as much weight to a match in the name field as the hobby field.
+
+== *Q: Why is indexing so slow?*
+
+By default, FTS tries to index all the fields in a document yet you typically only need to search on a few key fields.
+Indexing all fields requires more computation and results in larger indexes than you get if you limit the index to just the fields you need.
+
+Here are some tips to speed up indexing:
+
+* Build a custom mapping, set "dynamic": false to disable automatic indexing, and explicitly index only the named fields that you need to search.
+* Avoid numeric/date fields if you do not plan to query on these fields.
```

Comment:
Avoid numeric/date fields if you do not plan to query on these fields. -> Double negation avoid, not.

* Avoid querying on numeric or date fields.

Reply:
This changes the meaning intented here.

Original intent was this - user should avoid indexing numeric/date fields if they dont have query use case/requirements around those non-text fields.

Reply:
Updated